lib/IRGen/GenDecl.cpp

```diff
@@ -2910,6 +2910,11 @@ IRGenModule::getAddrOfLLVMVariableOrGOTEquivalent(LinkEntity entity) {
     return {gotEquivalent, ConstantReference::Indirect};
   };
   
+  // Avoid to create GOT because wasm32 doesn't support
+  // dynamic linking yet
+  if (TargetInfo.OutputObjectFormat == llvm::Triple::Wasm) {
+    return direct();
```

Comment:
It doesn't seem safe to always create direct references unless you're in WMO or one of the other checks for direct access succeeds. If the platform doesn't fully support non-WMO, perhaps we should error at the driver level to avoid miscompiling and throw an assertion down here.

Reply:
Do you mean there are some places that assume indirect reference when non-WMO?

Reply:
Ah, seems like WASM currently requires everything to be bundled into the same linkage unit. As long as you’re building the world, this is fine.